### PR TITLE
#4844 feat: renew id token when expires

### LINKED ISSUE #4844: Offer users to reconnect when id_token expires
The following error occurs when a JWT token has expired when requesting for `api/account/get`. 

I think one way to address this is by going to FlowCrypt settings -> resetting the account -> sign back in to refresh the id_token needed by the backend (not user friendly). 

The customer wonders if there's something we can do to simplify it. 

Reference: https://mail.google.com/mail/u/human@flowcrypt.com/#inbox/FMfcgzGrbbxhKWXXkjFGnRLcMxlLkRQc

### PR DESCRIPTION
**Pull request by ioanmo226**
This PR <what does it do and why?>

close #4844 // if this PR closes an issue

----------------------------------

**Tests** _(delete all except exactly one)_:
- Difficult to test (Hard/impossible to reproduce in semaphoreci)

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [x] addresses the issue it closes (if any)
- [x] code is readable and understandable
- [x] is accompanied with tests, or tests are not needed
- [x] is free of vulnerabilities
- [x] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

File: extension/js/common/api/account-servers/external-service.ts [hunk 1]
```diff
@@ -171,18 +172,45 @@ export class ExternalService extends Api {
...
+          ...(await this.authHdr()),
+        },
+        'json',
+        method
+      );
+    } catch (firstAttemptErr) {
+      const idToken = await InMemoryStore.get(this.acctEmail, InMemoryStoreKeys.ID_TOKEN);
+      if (ApiErr.isAuthErr(firstAttemptErr) && idToken) {
+        // force refresh token
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```
Comment (sosnovsky) [inline extension/js/common/api/account-servers/external-service.ts:193]: It seems `eslint-disable` not needed here

File: extension/js/common/api/account-servers/external-service.ts [hunk 2]
```diff
@@ -171,18 +172,45 @@ export class ExternalService extends Api {
...
+        const { email } = GoogleAuth.parseIdToken(idToken);
+        if (email) {
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```
Comment (sosnovsky) [inline extension/js/common/api/account-servers/external-service.ts:196]: And here too

**Review by sosnovsky — COMMENTED** [review]
This solution is much simpler than the previous one, looks great 👍
Just need to remove a couple unnecessary `eslint-disable` comments

File: extension/js/common/api/account-servers/external-service.ts (hunk 1 above)
Comment (ioanmo226) [reply extension/js/common/api/account-servers/external-service.ts:193]: Ok

**Review by sosnovsky — APPROVED** [review]
Looks good 👍

I'll create another issue for testing id tokens refresh, maybe it's possible to reproduce with our mock apis.

**Pull request merged**